scaffoldmaker/meshtypes/meshtype_3d_colon1.py

```diff
@@ -0,0 +1,448 @@
+            haustraRadiusFraction, foldFactor, haustraCurvatureFactor, wallThickness, haustraLength)
+
+        # Generate tube mesh
+        nextNodeIdentifier, nextElementIdentifier = generatetubemesh(region, elementsCountAround, elementsCountAlongHaustra, elementsCountThroughWall, haustraCount,
+            cx, cd1, xHaustraOuter, d1HaustraOuter, d2HaustraOuter, wallThickness, haustraAxis, haustraLength, useCrossDerivatives, useCubicHermiteThroughWall)
+
+    @classmethod
+    def generateMesh(cls, region, options):
+        """
+        Generate base or refined mesh.
+        :param region: Zinc region to create mesh in. Must be empty.
+        :param options: Dict containing options. See getDefaultOptions().
+        """
+        if not options['Refine']:
+            cls.generateBaseMesh(region, options)
+            return
+
+        refineElementsCountAround = options['Refine number of elements around']
+        refineElementsCountAlong = options['Refine number of elements along']
+        refineElementsCountThroughWall = options['Refine number of elements through wall']
+
+        baseRegion = region.createRegion()
+        cls.generateBaseMesh(baseRegion, options)
+
+        meshrefinement = MeshRefinement(baseRegion, region)
+        meshrefinement.refineAllElementsCubeStandard3d(refineElementsCountAround, refineElementsCountAlong, refineElementsCountThroughWall)
+
+def getUnitHaustraOuter(elementsCountAround, elementsCountAlongHaustra, radius, cornerRadiusFraction,
+        haustraRadiusFraction, foldFactor, haustraCurvatureFactor, wallThickness, haustraLength):
+    """
+    Generates a 3-D haustra unit mesh with variable numbers
+    of elements around, along the central line, and through wall.
+    The haustra has a triangular profile with rounded corners at the
+    interhaustral septa, and a clover profile outside the septa.
+    :param elementsCountAround: Number of elements around haustra.
+    :param elementsCountAlongHaustra: Number of elements along haustra.
+    :param radius: Inner radius of haustra.
+    :param cornerRadiusFraction: Roundness of triangular corners of
+    interhaustral septa. Factor is multiplied by inner haustra radius
+    to get a radius of curvature at the corners.
+    :param haustraRadiusFraction: Factor is multiplied by inner haustra
+    radius to obtain radius of intersecting circles in the middle cross-section
+    along a haustra.
+    :param foldFactor: Factor is multiplied by haustra length to scale derivative
+    along the haustra at the interhaustral septa end.
+    :param haustraCurvatureFactor: Factor is multiplied by haustra length to 
+    scale derivative along the haustra at the middle section of the haustra.
+    :param wallThickness: Thickness of haustra through wall.
+    :param haustraLength: Length of a haustra unit.
+    :return: coordinates, derivatives on outer surface of haustra unit.
```

Comment:
:return: xOuter, d1Outer, d2Outer, unitZ
What is unitZ needed for? Can easily get from cross product?